benchmark/clblast/blas1/asum.cpp

```diff
@@ -49,6 +49,25 @@ void run(benchmark::State& state, ExecutorType* executorPtr, index_t size) {
   MemBuffer<scalar_t, CL_MEM_WRITE_ONLY> buf1(executorPtr, v1.data(), size);
   MemBuffer<scalar_t, CL_MEM_READ_ONLY> bufr(executorPtr, &vr, 1);
 
+#ifdef BLAS_VERIFY_BENCHMARK
+  // Run a first time with a verification of the results
+  scalar_t vr_ref = reference_blas::asum(size, v1.data(), 1);
+  scalar_t vr_temp = 0;
+  {
+    MemBuffer<scalar_t, CL_MEM_READ_ONLY> vr_temp_gpu(executorPtr, &vr_temp, 1);
+    cl_event event;
+    clblast::Asum<scalar_t>(size, vr_temp_gpu.dev(), 0, buf1.dev(), 0, 1,
+                            executorPtr->_queue(), &event);
+    CLEventHandler::wait(event);
+  }
+
+  if (!utils::almost_equal<scalar_t>(vr_temp, vr_ref)) {
+    std::cerr << "Value mismatch: " << vr_temp << "; expected " << vr_ref
+              << std::endl;
+    exit(1);
```

Comment:
use assert of exit(1)

Reply:
As discussed offline, `exit` is the best thing to use here.

Reply:
```

Reply:
`assert` can be disable by some flags, and `abort` doesn't call the `atexit` handler